### PR TITLE
Tweaks the Aurora Caelus event 

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Tweaks the colors of space during the Aurora Caelus event. Also adds a positive mood associated with seeing the stars, as the event flavor text suggests that employees take this time off and relax.
For the sake of optimization, this moodlet is given to everyone on the station Z level when the event starts, instead of constantly checking if someone is seeing a lit up space tile to give it to them.

## Why It's Good For The Game
The event gets starlight colors that reflect the polar aurora lights. And now people will actually be happy when it happens because of the positive mood event.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Sped up version : 

https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/d0796940-10ec-4b37-81df-cb8adb14ac32

</details>

## Changelog
:cl:
add: Added a new mood event associated with the Aurora Caelus event
tweak: Tweaked the colors of the Aurora Caelus event
/:cl:

### CODE REVIEW

**Comment by TerryTehCoder** [comment]
Hasn't this been broken for a loong time? I feel I've never actually seen what it was supposed to look like.

**Comment by CydiaLamiales** [comment]
> Hasn't this been broken for a loong time? I feel I've never actually seen what it was supposed to look like.

It was, it creates a light with 3x the light radius of a previous light on a space tile... except space tiles never had any light to begin with, so it doesn't create one.

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]
This needs to include fake space tiles

**Comment by CydiaLamiales** [comment]
Will add that when I get home

File: code/modules/events/aurora_caelus.dm [hunk 1]
```diff
@@ -12,51 +12,68 @@
...
+		if(is_station_level(M.z))
+			SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "star_gazing", /datum/mood_event/witnessed_starlight)
+			if(M.client.prefs.toggles & PREFTOGGLE_SOUND_MIDI)
+				M.playsound_local(M, 'sound/ambience/aurora_caelus.ogg', 20, FALSE, pressure_affected = FALSE)
+	for(var/atom/window in GLOB.station_windows)
+		for(var/turf/T as() in RANGE_TURFS(1, window))
+			if(isspaceturf(T) && !(T.starlit))
+				T.set_light(1.5, 0.1, aurora_colors[7])
+				T.starlit = TRUE
+				affected_turfs += T
```
Comment (EvilDragonfiend) [inline code/modules/events/aurora_caelus.dm]: this should be in a separate proc.

File: code/modules/events/aurora_caelus.dm [hunk 2]
```diff
@@ -12,51 +12,68 @@
...
-				for(var/turf/open/space/S in A)
-					S.set_light(l_color = aurora_color)
+		var/light_modifier = 0
+		if(aurora_progress < 5)
+			light_modifier = aurora_progress / 10
+		else
+			light_modifier = 1 - aurora_progress / 10
+		for(var/atom/A in affected_turfs)
+			A.set_light(2 + light_modifier * 4, 0.3, l_color = aurora_color)
+
```
Comment (EvilDragonfiend) [inline code/modules/events/aurora_caelus.dm]: the whole codes in tick() here should be in a separate proc, and that proc should be called by ASYNC() from tick()

File: code/modules/events/aurora_caelus.dm [hunk 3]
```diff
@@ -12,51 +12,68 @@
...
 	sender_override = "Nanotrasen Meteorology Division")
+	starttime = world.timeofday
 	for(var/V in GLOB.player_list)
 		var/mob/M = V
-		if((M.client.prefs.toggles & PREFTOGGLE_SOUND_MIDI) && is_station_level(M.z))
-			M.playsound_local(M, 'sound/ambience/aurora_caelus.ogg', 20, FALSE, pressure_affected = FALSE)
+		if(is_station_level(M.z))
+			SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "star_gazing", /datum/mood_event/witnessed_starlight)
+			if(M.client.prefs.toggles & PREFTOGGLE_SOUND_MIDI)
+				M.playsound_local(M, 'sound/ambience/aurora_caelus.ogg', 20, FALSE, pressure_affected = FALSE)
```
Comment (EvilDragonfiend) [inline code/modules/events/aurora_caelus.dm]: this should be in a separated proc too.

**Review by EvilDragonfiend — DISMISSED** [review]
This event is a bit heavy because it handles too many atoms, so that it should use ASYNC. I will tell you what you should do further once these change requests are addressed.

**Comment by PowerfulBacon** [comment]
This will be depreciated by the starlight rework, do you want to keep the mood changes?

**Comment by CydiaLamiales** [comment]
> This will be depreciated by the starlight rework, do you want to keep the mood changes?

Mood and color changes, yes.

**Comment by CydiaLamiales** [comment]
Well, since Bacon basically "fixed" the event by adding starlight in another PR, this will instead just be a color tweak + mood addition.

File: code/game/turfs/turf.dm [hunk 4]
```diff
@@ -60,6 +60,7 @@ GLOBAL_LIST_EMPTY(created_baseturf_lists)
 	///Icon-smoothing variable to map a diagonal wall corner with a fixed underlay.
 	var/list/fixed_underlay = null
 
+
```
Comment (CydiaLamiales) [inline code/game/turfs/turf.dm]:
```suggestion
```

File: code/game/objects/structures/window.dm [hunk 5]
```diff
@@ -70,7 +70,6 @@
 	var/static/list/loc_connections = list(
 		COMSIG_ATOM_EXIT = PROC_REF(on_exit),
 	)
```
Comment (CydiaLamiales) [inline code/game/objects/structures/window.dm]:
```suggestion
	)

```

**Review by Rukofamicom — APPROVED** [review]

**Pull request merged**